web/src/main/java/com/synopsys/integration/alert/web/api/provider/ProviderNameSelectCustomFunctionAction.java

```diff
@@ -18,36 +16,25 @@
 import com.synopsys.integration.alert.common.action.ActionResponse;
 import com.synopsys.integration.alert.common.action.CustomFunctionAction;
 import com.synopsys.integration.alert.common.descriptor.DescriptorMap;
-import com.synopsys.integration.alert.common.descriptor.config.field.LabelValueSelectOption;
 import com.synopsys.integration.alert.common.descriptor.config.field.LabelValueSelectOptions;
 import com.synopsys.integration.alert.common.descriptor.config.field.validation.FieldValidationUtility;
 import com.synopsys.integration.alert.common.descriptor.config.ui.ChannelDistributionUIConfig;
-import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
-import com.synopsys.integration.alert.common.enumeration.DescriptorType;
 import com.synopsys.integration.alert.common.rest.HttpServletContentWrapper;
 import com.synopsys.integration.alert.common.rest.model.FieldModel;
 import com.synopsys.integration.alert.common.security.authorization.AuthorizationManager;
 
 @Component
 public class ProviderNameSelectCustomFunctionAction extends CustomFunctionAction<LabelValueSelectOptions> {
-    private final DescriptorMap descriptorMap;
 
     @Autowired
     public ProviderNameSelectCustomFunctionAction(AuthorizationManager authorizationManager, DescriptorMap descriptorMap, FieldValidationUtility fieldValidationUtility) {
         super(ChannelDistributionUIConfig.KEY_PROVIDER_NAME, authorizationManager, descriptorMap, fieldValidationUtility);
-        this.descriptorMap = descriptorMap;
     }
 
     @Override
     public ActionResponse<LabelValueSelectOptions> createActionResponse(FieldModel fieldModel, HttpServletContentWrapper servletContentWrapper) {
-        List<LabelValueSelectOption> options = descriptorMap.getDescriptorByType(DescriptorType.PROVIDER).stream()
-                                                   .map(descriptor -> descriptor.createMetaData(ConfigContextEnum.DISTRIBUTION))
-                                                   .flatMap(Optional::stream)
-                                                   .map(descriptorMetadata -> new LabelValueSelectOption(descriptorMetadata.getLabel(), descriptorMetadata.getName()))
-                                                   .sorted()
-                                                   .collect(Collectors.toList());
-        LabelValueSelectOptions optionList = new LabelValueSelectOptions(options);
-        return new ActionResponse<>(HttpStatus.OK, optionList);
+        // FIXME removed this as it's no longer being used. Should be deleted before release
+        return new ActionResponse(HttpStatus.OK, List.of());
```

Comment:
We should Deprecate this method, and we can change the response to an error code rather than OK.

Reply:
Should we deprecate most custom function controllers this version since they will most likely be removed in the future?